From 79a9d9061c57583078ec6a43f5b6a1482b573e76 Mon Sep 17 00:00:00 2001
From: Pablo Rodriguez <pablo.rodriguez-quesada@windriver.com>
Date: Thu, 19 Mar 2020 20:04:00 -0600
Subject: [PATCH 1/3] arm64: add support for arm64 architectures

There are many other architectures that can benefit of this plugin.
The jetson family for instance can make use of this plugin.

Signed-off-by: Pablo Rodriguez <pablo.rodriguez-quesada@windriver.com>
---
 docker/arm64/Dockerfile.ubuntu16.04 | 43 +++++++++++++++++++++++++++++
 1 file changed, 43 insertions(+)
 create mode 100644 docker/arm64/Dockerfile.ubuntu16.04

diff --git a/docker/arm64/Dockerfile.ubuntu16.04 b/docker/arm64/Dockerfile.ubuntu16.04
new file mode 100644
index 00000000..3faf20ce
--- /dev/null
+++ b/docker/arm64/Dockerfile.ubuntu16.04
@@ -0,0 +1,43 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM --platform=arm64 ubuntu:16.04 as build
+
+RUN apt-get update && apt-get install -y --no-install-recommends \
+        g++ \
+        ca-certificates \
+        wget && \
+    rm -rf /var/lib/apt/lists/*
+
+ENV GOLANG_VERSION 1.10.3
+RUN wget -nv -O - https://storage.googleapis.com/golang/go${GOLANG_VERSION}.linux-arm64.tar.gz \
+    | tar -C /usr/local -xz
+ENV GOPATH /go
+ENV PATH $GOPATH/bin:/usr/local/go/bin:$PATH
+
+WORKDIR /go/src/nvidia-device-plugin
+COPY . .
+
+RUN export CGO_LDFLAGS_ALLOW='-Wl,--unresolved-symbols=ignore-in-object-files' && \
+    go install -ldflags="-s -w" -v nvidia-device-plugin
+
+
+FROM --platform=arm64 debian:stretch-slim
+
+ENV NVIDIA_VISIBLE_DEVICES=all
+ENV NVIDIA_DRIVER_CAPABILITIES=utility
+
+COPY --from=build /go/bin/nvidia-device-plugin /usr/bin/nvidia-device-plugin
+
+CMD ["nvidia-device-plugin"]
-- 
2.17.1

